sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_arg.asciidoc

```diff
@@ -366,6 +366,86 @@ a compile error if `PropertyListT` does not contain a property with the
 Available only if `propertyT` is the property key class of a compile-time
 constant property.
 
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList>
+auto operator+(const annotated_arg<T2, PropertyList> &other) const;
```

Comment:
Several questions:

* What is the return type for these operators?  Just saying `auto` isn't very helpful.  In particular, is the return type `T` or `annotated_ptr<T>`?  If the latter, what are the properties?
* These operators all assume the RHS is `annotated_arg`.  What if the user wants to do `operator+` where the RHS is a raw type without `annotated_arg`?
* What are the constraints on the type `T2`?
* Can't we just rely on the implicit conversion from `annotated_arg<T>` to `T`, rather than adding all these operators?

Reply:
> Can't we just rely on the implicit conversion from annotated_arg<T> to T, rather than adding all these operators?

Greg, see my comment on the PR (also added to description) regarding this one.

Reply:
> What is the return type for these operators? Just saying auto isn't very helpful.

I think the return type necessarily has to be auto if we want to keep this generic. For example take the case where `annotated_arg` is templated on Intel's `ac_int` library then multiplication returns a wider type the size of which is decided by the `ac_int::operator*`.

> In particular, is the return type T or annotated_ptr<T>? If the latter, what are the properties?

Returned type would not be an `annotated_arg`

Reply:
I think it would help to specify the constraints and the return type better.  For example:

> Available only if `operator+(T2)` is valid for objects of type `T`.  The value and type of the result is the same as the result of `operator+(T2)` applied to the object of type `T`.

Reply:
> I tested this and it would also require additional overloaded operators with a generic type such as this:

Use SFINAE:
```
template<typename T, typename T2, typename R = decltype(std::declval<T>() + std::declval<T2>())>
R operator+(annotated_arg<T> a, T2 b) {
    return a + b;
};
```
I think this will allow you to use an object of type annotated_arg<T> in any operator+ in which you could have used an object of type T.

Reply:
@GarveyJoe I tested this notation and the following code works: 
```cpp
template<typename T, typename PropertyList, typename T2, typename R = decltype(std::declval<T>() + std::declval<T2>())>
R operator+(const annotated_arg<T, PropertyList> &a, T2 b) {
    T a1 = a;
    return a1 + b;
};
```

Can you confirm that we want to add these operators in addition to the operators I've currently added for `annotated_arg<T1> + annotated_arg<T2>`?

Reply:
Yes, confirmed